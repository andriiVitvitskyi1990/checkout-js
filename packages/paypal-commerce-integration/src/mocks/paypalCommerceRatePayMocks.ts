import {
    Cart,
    Checkout,
    Consignment,
    Coupon,
    Customer,
    Discount,
    GiftCertificate,
    Tax,
} from '@bigcommerce/checkout-sdk';

export function getPaypalCommerceRatePayMethodMock() {
    return {
        clientToken: 'token',
        config: {
            cardCode: true,
            displayName: 'RatePay',
            enablePaypal: undefined,
            hasDefaultStoredInstrument: false,
            helpText: '',
            is3dsEnabled: undefined,
            isHostedFormEnabled: false,
            isVaultingCvvEnabled: undefined,
            isVaultingEnabled: false,
            isVisaCheckoutEnabled: undefined,
            logo: undefined,
            merchantId: '',
            requireCustomerCode: false,
            testMode: true,
        },
        gateway: 'paypalcommercealternativemethods',
        id: 'ratepay',
        initializationData: {
            attributionId: 'BC_id',
            availableAlternativePaymentMethods: [],
            clientId: 'asd123',
            enabledLocalPaymentMethods: [],
            intent: 'capture',
            isComplete: false,
            isDeveloperModeApplicable: false,
            isHostedCheckoutEnabled: true,
            isInlineCheckoutEnabled: false,
            isPayPalCreditAvailable: true,
            isVenmoEnabled: true,
            merchantId: 'XdasdQWe1123',
            orderId: undefined,
            shouldRenderFields: true,
            showOnlyOnMobileDevices: false,
        },
        initializationStrategy: {
            type: 'not_applicable',
        },
        logoUrl: 'http://logo_url_path',
        method: 'braintree',
        nonce: undefined,
        returnUrl: undefined,
        supportedCards: [
            'PAYPAL',
            'PAYPALCREDIT',
            'VISA',
            'MC',
            'AMEX',
            'DISCOVER',
            'VENMO',
            'APPLEPAY',
        ],
        type: 'PAYMENT_TYPE_API',
    };
}

export function getCheckoutMock(): Checkout {
    return {
        id: '1',
        // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
        cart: {
            currency: {
                code: 'USD',
                decimalPlaces: 2,
            },
        } as Cart,
        // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
        customer: {} as Customer,
        customerMessage: '',
        // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
        consignments: [{}] as Consignment[],
        // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
        taxes: [{}] as Tax[],
        // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
        discounts: [{}] as Discount[],
        // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
        coupons: [{}] as Coupon[],
        isStoreCreditApplied: false,
        shippingCostTotal: 0,
        shippingCostBeforeDiscount: 0,
        shouldExecuteSpamCheck: false,
        handlingCostTotal: 0,
        outstandingBalance: 1,
        taxTotal: 0,
        subtotal: 0,
        grandTotal: 1,
        // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
        giftCertificates: [{}] as GiftCertificate[],
        balanceDue: 0,
        createdTime: '',
        updatedTime: '',
        giftWrappingCostTotal: 0,
        channelId: 1,
    };
}
