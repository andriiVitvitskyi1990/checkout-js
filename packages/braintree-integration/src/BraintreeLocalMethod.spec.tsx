import { createCheckoutService, LanguageService } from '@bigcommerce/checkout-sdk';
import { render } from '@testing-library/react';
import React from 'react';

import { PaymentFormService } from '@bigcommerce/checkout/payment-integration-api';

import { getBraintreeLocalMethodMock } from './mocks/braintreeLocalMethodsMocks';
import BraintreeLocalMethod from './BraintreeLocalPaymentMethod';

describe('BraintreeLocalMethod', () => {
    const checkoutService = createCheckoutService();
    const checkoutState = checkoutService.getState();
    const props = {
        method: getBraintreeLocalMethodMock(),
        checkoutService,
        checkoutState,
        // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
        paymentForm: jest.fn() as unknown as PaymentFormService,
        // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
        language: { translate: jest.fn() } as unknown as LanguageService,
        onUnhandledError: jest.fn(),
    };

    it('renders component with provided props', () => {
        const { container } = render(<BraintreeLocalMethod {...props} />);

        expect(container).toMatchSnapshot();
    });
});
