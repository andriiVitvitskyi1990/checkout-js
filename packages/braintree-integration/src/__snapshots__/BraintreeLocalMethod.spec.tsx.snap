// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BraintreeLocalMethod renders component with provided props 1`] = `<div />`;
